compiler-tests/src/test/java/com/squareup/anvil/compiler/dagger/InjectConstructorFactoryGeneratorTest.kt

```diff
@@ -0,0 +1,2819 @@
+          InjectClass_Factory<T>(param0)
+
+      @JvmStatic
+      public fun <T : List<String>> newInstance(param0: T): InjectClass<T> = InjectClass<T>(param0)
+    }
+  }
+       */
+
+      skipWhen(mode.isK2) { "https://github.com/square/anvil/issues/1122" }
+
+      compile2(
+        """
+        package com.squareup.test
+        
+        import javax.inject.Inject
+        import javax.inject.Provider
+  
+        class InjectClass<T : List<String>> @Inject constructor(
+          private val t: T
+        )
+        """,
+      ) {
+
+        val factoryClass = classLoader.injectClass_Factory
+
+        val typeParams = factoryClass.typeParameters
+          .associate { param ->
+            param.name to param.bounds.map { it.typeName }
+          }
+
+        assertThat(typeParams)
+          .containsExactly("T", listOf("java.util.List<? extends java.lang.String>"))
+
+        val constructor = factoryClass.declaredConstructors.single()
+        assertThat(constructor.parameterTypes.toList())
+          .containsExactly(Provider::class.java)
+
+        val staticMethods = factoryClass.declaredMethods.filter { it.isStatic }
+
+        val factoryInstance = staticMethods.single { it.name == "create" }
+          .invoke(null, Provider { listOf<String>() })
+
+        assertThat(factoryInstance::class.java).isEqualTo(factoryClass)
+
+        val instance = staticMethods.single { it.name == "newInstance" }
+          .invoke(null, listOf<String>())
+
+        assertThat(instance).isNotNull()
+        assertThat((factoryInstance as Factory<*>).get()).isNotNull()
+      }
+    }
+
+  @TestFactory
+  fun `a factory class is generated for an inject constructor without a package`() =
+    testFactory {
+      compile2(
+        """
+      import javax.inject.Inject
+      
+      class InjectClass @Inject constructor()
+      """,
+      ) {
+        val factoryClass = classLoader.loadClass("InjectClass").factoryClass()
+
+        val constructor = factoryClass.declaredConstructors.single()
+        assertThat(constructor.parameterTypes.toList()).isEmpty()
+
+        val staticMethods = factoryClass.declaredMethods.filter { it.isStatic }
+
+        val factoryInstance = staticMethods.single { it.name == "create" }
+          .invoke(null)
+        assertThat(factoryInstance::class.java).isEqualTo(factoryClass)
+
+        val instance = staticMethods.single { it.name == "newInstance" }
+          .invoke(null)
+
+        assertThat(instance).isNotNull()
+        assertThat((factoryInstance as Factory<*>).get()).isNotNull()
+      }
+    }
+
+  @TestFactory
+  fun `two inject constructors aren't supported`() = testFactory {
+    compile2(
+      """
+      package com.squareup.test
+      
+      import javax.inject.Inject
+      
+      class InjectClass @Inject constructor() {
+        @Inject constructor(string: String)
+      }
+      """,
+      expectedExitCode = ExitCode.COMPILATION_ERROR,
+    ) {
+      // assertThat(
```

Comment:
This test is showing as commented out for me still, leaving a comment in case that was meant to be included in the last push. Otherwise everything looks good!

Reply:
I'll email you a zip file with the change.